lib/collector/errors.js

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright 2024 New Relic Corporation. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+'use strict'
+
+const ERR_CODES = {
```

Comment:
not sure we need all of this for this ticket. i would think you're only handling the max payload exceeded

Reply:
I figured it might be a start to centralizing errors and establishing error codes (similar to https://github.com/fastify/fastify/blob/ffbc92c78a588e5ec6f16d20492f23b08654345f/lib/errors.js). But I'm not tied to this. I can make the payload exceeded error unique if you think that would be better.

Reply:
it may be fine but would prefer we keep this PR scoped to fixing the bug. especially since the other codes are errors on construction of a remote method class which only happens at boostrap of agent or when we make a request but it's all http options that we are controlling